gc.c

```diff
@@ -1817,7 +1818,10 @@ rb_objspace_free(rb_objspace_t *objspace)
     }
     st_free_table(objspace->id_to_obj_tbl);
     st_free_table(objspace->obj_to_id_tbl);
+
     free_stack_chunks(&objspace->mark_stack);
+    mark_stack_free_cache(&objspace->mark_stack);
```

Comment:
Seems calling `mark_stack_chunk_list_free` twice, in both calls.

Reply:
`free_stack_chunks` is calling `mark_stack_chunk_list_free` on `stack->chunk` while `mark_stack_free_cache` uses `stack->cache`. They're freeing two different linked lists.